chord_drs/backends/local.py

```diff
@@ -19,14 +24,50 @@ def __init__(self, config: dict):  # config is dict or flask.Config, which is a
         # We can use mkdir, since resolve has been called in config.py
         self.base_location.mkdir(parents=True, exist_ok=True)
 
-    def save(self, current_location: str | Path, filename: str) -> str:
+    async def save(self, current_location: str | Path, filename: str) -> str:
         new_location = self.base_location / filename
         copy(current_location, new_location)
         return str(new_location.resolve())
 
-    def delete(self, location: str | Path) -> None:
+    async def delete(self, location: str | Path) -> None:
         loc = location if isinstance(location, Path) else Path(location)
         if self.base_location in loc.parents:
             loc.unlink()
             return
         raise ValueError(f"Location {loc} is not a subpath of backend base location {self.base_location}")
+
+    async def get_stream_generator(self, location: str, range: tuple[int, int]) -> Generator[Any, None, None]:
+        if range:
+            start, end = range
+            generator = self._stream_range(location, start, end)
+        else:
+            generator = self._stream_whole(location)
+        return sync_generator_stream(generator)
+
+    async def _stream_range(self, location: str, start: int, end: int):
```

Comment:
this seems more or less identical to https://github.com/bento-platform/bento_lib/blob/master/bento_lib/streaming/file.py – is there any way we can consolidate this to the `bento_lib` version? (possibly with changes to `bento_lib`)

Reply:
also, add return type here (should be a generator of `bytes`)